Application/SolutionScripts/InitializeDevelopmentEnvironment.psm1

```diff
@@ -250,51 +250,58 @@ Set-Alias -Scope Global Rebuild-Solution Invoke-RebuildSolution
 Function Invoke-RebuildSolution {
     Param(
         [string] $buildConfiguration = "Debug",
-        [string] $target = "build",
         [string] $verbosity = "minimal",
         [string] $maxCpuCount = "",
         [string] $noReuse = "false",
-        [string] $platformEnvironmentVariable = "Any CPU",
-        [string] $solutionPath = (Get-RepositoryResolvedPath "Application/Ed-Fi-Ods.sln"),
-        [string] $msBuildFilePath = $env:msbuild_exe
+        [string] $solutionPath = (Get-RepositoryResolvedPath "Application/Ed-Fi-Ods.sln")
     )
     Invoke-Task -name $MyInvocation.MyCommand.Name -task {
         if ((Get-DeploymentSettings).Engine -eq 'PostgreSQL') { $buildConfiguration = 'Npgsql' }
         if (-not [string]::IsNullOrWhiteSpace($env:msbuild_buildConfiguration)) { $buildConfiguration = $env:msbuild_buildConfiguration }
 
-        $msBuildParameters = @(
-            "`"/property:Configuration=$buildConfiguration`"",
-            "`"/target:$target`"",
-            "`"/verbosity:$verbosity`"",
-            "`"/nr:$noReuse`"",
-            "`"/p:Platform=$platformEnvironmentVariable`""
-        )
-
         if (-not [string]::IsNullOrWhiteSpace($maxCpuCount)) {
-            $msBuildParameters += "`"/m:$maxCpuCount`""
+            $maxCpuCount = "/m:$maxCpuCount"
         }
         else {
-            $msBuildParameters += "`"/m`""
+            $maxCpuCount = "/m"
         }
 
         $params = @{
             Path                           = $solutionPath
-            MsBuildParameters              = ($msBuildParameters -join ' ')
-            ShowBuildOutputInCurrentWindow = $true
+            BuildConfiguration             = $buildConfiguration
             LogVerbosityLevel              = $verbosity
         }
 
-        if (-not [string]::IsNullOrWhiteSpace($msBuildFilePath)) { $params.MsBuildFilePath = $msBuildFilePath }
-
         ($params).GetEnumerator() | Sort-Object -Property Name | Format-Table -HideTableHeaders -AutoSize -Wrap | Out-Host
-        $buildResult = Invoke-MsBuild @params
 
-        if ($null -eq $buildResult.BuildSucceeded) { throw "Unsure if build passed or failed: $($buildResult.Message)" }
-        if ($buildResult.BuildSucceeded -eq $true) { return }
-        if ($buildResult.BuildSucceeded -eq $false) {
+        # Local Variables.
+        $BuildLogDirectoryPath = $env:Temp
+        
+        # Always get the full path to the Log files directory.
+        $BuildLogDirectoryPath = [System.IO.Path]::GetFullPath($BuildLogDirectoryPath)
+        
+        # Local Variables.
+        $solutionFileName = (Get-ItemProperty -LiteralPath $solutionPath).Name
+        $buildLogFilePath = (Join-Path -Path $BuildLogDirectoryPath -ChildPath $solutionFileName) + ".msbuild.log"
+        
+        # Build
+        dotnet build $solutionPath -c $buildConfiguration -v $verbosity $maxCpuCount /nr:$noReuse /flp:v=$verbosity /flp:logfile=$buildLogFilePath | Out-Host
```

Comment:
i am concerned here looks like parameters are not supported. Should we continue with max cpus and noderesuse?
```
C:\Projects\Ed-Fi-MigrationUtility [ODS-4740]> dotnet build --help
Usage: dotnet build [options] <PROJECT | SOLUTION>

Arguments:
  <PROJECT | SOLUTION>   The project or solution file to operate on. If a file is not specified, the command will search the current directory for one.

Options:
  -h, --help                            Show command line help.
  -o, --output <OUTPUT_DIR>             The output directory to place built artifacts in.
  -f, --framework <FRAMEWORK>           The target framework to build for. The target framework must also be specified in the project file.
  -c, --configuration <CONFIGURATION>   The configuration to use for building the project. The default for most projects is 'Debug'.
  -r, --runtime <RUNTIME_IDENTIFIER>    The target runtime to build for.
  --version-suffix <VERSION_SUFFIX>     Set the value of the $(VersionSuffix) property to use when building the project.
  --no-incremental                      Do not use incremental building.
  --no-dependencies                     Do not build project-to-project references and only build the specified project.
  /nologo, --nologo                     Do not display the startup banner or the copyright message.
  --no-restore                          Do not restore the project before building.
  --interactive                         Allows the command to stop and wait for user input or action (for example to complete authentication).
  -v, --verbosity <LEVEL>               Set the MSBuild verbosity level. Allowed values are q[uiet], m[inimal], n[ormal], d[etailed], and diag[nostic].
  --force                               Force all dependencies to be resolved even if the last restore was successful.
                                        This is equivalent to deleting project.assets.json.
C:\Projects\Ed-Fi-MigrationUtility [ODS-4740]>
```

Reply:
you may want to look at this to make max cpu work https://www.hanselman.com/blog/faster-builds-with-msbuild-using-parallel-builds-and-multicore-cpus

Reply:
https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-build

Reply:
Removed both